### PR TITLE
v1.18: Re-enable SPL downstream job with last v1.18 commit

### PR DESCRIPTION
**Pull request by joncinque**
#### Problem

https://github.com/anza-xyz/agave/pull/1961/files did a great job with avoiding issues on the downstream build. However, it means that we might not catch a breaking change during a backport.

#### Summary of Changes

Checkout the v1.18 branch from SPL, which is tagged to the last commit using v1.18: https://github.com/solana-labs/solana-program-library/tree/v1.18, which will allow the job to resume for v1.18 backports

Fixes #

### CODE REVIEW

**Review by yihau — APPROVED** [review]
thank you!

**Comment by joncinque** [comment]
This PR is on top of the latest v1.18 commit, but is still failing the sanity step in CI -- is this normal? Should I wait for something before merging or merge on red?

**Comment by yihau** [comment]
should be okay to merge. just merged the version bumping PR 🫠

**Review by CriesofCarrots — APPROVED** [review]

**Pull request merged**
